web/src/pages/tester.tsx

```diff
@@ -0,0 +1,169 @@
+    initialQuestions.map((q, i) => ({ question: q, selected: true, index: i }))
+  );
+
+  const { settings, changeSetting, setMode, settingsLoaded } = useSettings();
+
+  /** Run a search for the given `question` and insert the query promise into it
+   */
+  const search = useCallback(
+    (
+      { question, sessionId, query, selected, index }: QuestionState,
+      controller: AbortController
+    ): QuestionState => ({
+      question,
+      sessionId,
+      selected,
+      index,
+      query: queryLLM(
+        question,
+        settings,
+        [],
+        updater(index),
+        sessionId,
+        controller
+      ),
+    }),
+    [settings]
+  );
+
+  useEffect(() => {
+    // Only run the initial search once the settings have been loaded from the URL and localstorage
+    if (!settingsLoaded) return;
+
+    updateQuestions((questions) =>
+      questions
+        .map((q) => ({ ...q, sessionId: q.sessionId || crypto.randomUUID() }))
+        .map((q) => (q.selected && !q.query ? search(q, controller) : q))
+    );
+  }, [settingsLoaded, controller, search]);
+
+  /** Return a function that will update the `index`-th question with the current results of its search
+   */
+  const updater = (index: number) => (entry: CurrentSearch) =>
+    updateQuestions((questions) => {
+      const question = questions[index];
+      if (!question) return questions;
+
+      questions[index] = { ...question, ...entry };
+      return [...questions];
+    });
+
+  const toggleQuestion = (i: number) => {
+    const question = questions[i];
+    if (question === undefined) return;
+
+    question.selected = !question.selected;
+
+    // If the question is to be displayed, but hasn't had a search run, run it now
+    if (question.selected && question.query === undefined) {
+      questions[i] = search(question, controller);
+    }
+    updateQuestions([...questions]);
+  };
+
+  const selectedQuestions = questions.filter(({ selected }) => selected);
+  return (
+    <Page page="tester" widescreen={true}>
+      <Controls mode={settings.mode || "default"} setMode={setMode} />
+      <div className="flex">
+        <div>
+          <ChatPrompts
+            settings={settings}
+            query="<this is where the query will go>"
+            history={[]}
+            changeSetting={changeSetting}
+          />
+          <div className="chat-settings mx-5 w-[400px] flex-none gap-4 border-2 outline-black">
+            {questions.map(({ question, selected }, i) => (
+              <div key={i} onClick={() => toggleQuestion(i)}>
+                <input
+                  type="checkbox"
+                  defaultChecked={selected}
+                  name={i.toString()}
+                  className="m-2"
+                />
+                <label htmlFor="i">{question}</label>
+              </div>
+            ))}
+            <button
+              onClick={() => {
+                controller.abort();
```

Comment:
is this component fast to re-render between a double click? if not, the same controller aborted in the first click will be aborted again and the one created in first click won't be aborted at all .. you might need a `useRef` for a mutable reference that might change multiple times per render..